userguide/tutorials/plugin_development.adoc

```diff
@@ -189,6 +189,57 @@ When you start developing your own plugin, it would be useful to be able to set
 
 . This runs the application in debug mode. You can also set additional breakpoints as required.
 
+=== Authentication within plugins
+
+In order to invoke read/write API operations like `AccountUserApi#getAccountById` , plugin code must authenticate against Kill Bill first. Otherwise, it will result in an `org.apache.shiro.authz.UnauthenticatedException`. This section explains how authentication can be done.
+
+==== Authentication steps
+
+In order to authenticate against Kill Bill, the following needs to be done:
+
+1. Obtain credentials - Typically, plugins cannot use the Kill Bill credentials and should be configured with a custom username/password. You can see an example of retrieving custom credentials in the https://github.com/killbill/killbill-email-notifications-plugin/blob/8df47156a2e80c65ce574e0ad689afd02b926f59/src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java#L230[EmailnotificationListener] code.
+
+2. Add the following code before invoking any API operation (use credentials obtained above):
+
+[source,java]
+killbillAPI.getSecurityApi().login(login, password);
+
+[start=3]
+3. Invoke the necessary api method:
+
+[source,java]
+invoiceUserApi.triggerDryRunInvoiceGeneration(accountId, targetDate, dryRunArguments, callContext);
+
+[start=4]
+4. Invoke the `logout` method:
+
+[source,java]
+osgiKillbillAPI.getSecurityApi().logout();
+
+==== Skipping Authentication
+
+It is also possible to skip authentication in the plugin code. For this, the following needs to be done:
+
+1. Set the following property in the config file:
```

Comment:
Let's remember to update the config document with this too!

Reply:
Yes, I have made a note of this.